api/libproject/devstart.rb

```diff
@@ -634,6 +634,116 @@ def run_local_all_migrations()
+  end
+  verificion_full_cmd
+end
+
+# Run a liqibase command against the specified project. Where possible, show SQL
+# statements and ask user for verification.
+def run_liquibase(cmd_name, *args)
+  command_to_sql = {
+      'changelogSync' => 'changelogSyncSQL',
+      'markNextChangesetRan' => 'markNextChangesetRanSQL',
+      'rollback' => 'rollbackSQL',
+      'rollbackCount' => 'rollbackCountSQL',
+      'rollbackToDate' => 'rollbackToDateSQL',
+      'update' => 'updateSQL',
+      'updateCount' => 'updateCountSql',
+      'updateToTag' => 'updateToTagSQL'
+  }
+
+  common = Common.new
+  ensure_docker(cmd_name, args)
+
+  op = WbOptionsParser.new(cmd_name, args)
+  op.add_typed_option(
+      "--command [command]",
+      String,
+      ->(opts, c) { opts.command = c},
+      "Liquibase command, e.g. update, rollback, tag, validate. See "+
+          "https://www.liquibase.org/documentation/command_line.html")
+  op.add_typed_option(
+      "--argument [argument]",
+      String,
+      ->(opts, a) { opts.argument = a},
+      "Liquibase command argument, e.g. count or tag value")
+  op.add_typed_option(
+        "--run-list [run_list]",
+        String,
+        ->(opts, rl) { opts.run_list = rl },
+        "Liquibase runList, a comma-separated list of activities in the liquibase task")
+  op.add_typed_option(
+        '--project [project]',
+        String,
+        ->(opts, p) { opts.project = p },
+        'AoU environment GCP project full name. Used to pick MySQL instance & creadentials.'
+  )
+  op.add_validator ->(opts) {
+    if opts.command.nil? || opts.command.empty?
+      raise ArgumentError.new("command is required")
+    end
+  }
+  op.parse.validate
+
+  # Currently there's only one activity (main), and leaving out the runList arugment causes
+  # it to run that activity. However, that's not because it's detected as default or primary, but
+  # leaving out the runList is equivalent to specifying all of the activities to run in unspecified
+  # order.
+  # https://github.com/liquibase/liquibase-gradle-plugin#3-configuring-the-plugin
+  if op.opts.run_list.nil? || op.opts.run_list.empty?
+    run_list = "main"
+  else
+    run_list = op.opts.run_list
+  end
+
+  if op.opts.argument.nil? || op.opts.argument.empty?
+    # It's safe to pass the '-PliquibaseCommand=' by itself, but it's not very clean
+    argument = ''
+  else
+    argument = "-PliquibaseCommandValue=#{op.opts.argument}"
+  end
+
+  if op.opts.project.nil? || op.opts.project.empty?
+    op.opts.project = 'local'
+  end
+
+  context = GcloudContextV2.new(op)
+  context.validate
+
+  with_optional_cloud_proxy_and_db(context, nil, 'sa-key.json') do |gcc|
+    common.status('inside with_optional_cloud_proxy_and_db')
+    common.status("project: #{gcc.project}, account: #{gcc.account}, creds_file: #{gcc.creds_file}, dir: #{Dir.pwd}")
+    command = op.opts.command
+
+    Dir.chdir('db') # 'cd' isn't available to run inline apparently
```

Comment:
I believe it executes a subprocess, presumably it didn't say `cd not found`, rather it didn't have any effect?

Reply:
If I change that line to `common.run_inline(%W{cd db})`, I get

```
project: local, account: jay.carlton@pmi-ops.org, creds_file: , dir: /w/api
+ cd db
Traceback (most recent call last):
	8: from /w/api/project.rb:6:in `<main>'
	7: from /w/aou-utils/workbench.rb:61:in `handle_argv_or_die'
	6: from /w/aou-utils/utils/common.rb:86:in `handle_or_die'
	5: from /w/api/libproject/devstart.rb:745:in `block in <top (required)>'
	4: from /w/api/libproject/devstart.rb:717:in `run_liquibase'
	3: from /w/api/libproject/devstart.rb:2117:in `with_optional_cloud_proxy_and_db'
	2: from /w/api/libproject/devstart.rb:723:in `block in run_liquibase'
	1: from /w/aou-utils/utils/common.rb:157:in `run_inline'
/w/aou-utils/utils/common.rb:157:in `spawn': No such file or directory - cd (Errno::ENOENT)
```

Reply:
Well.. consider me confused as well